include/libpmemfile-posix.h

```diff
@@ -0,0 +1,353 @@
+#define PMEMFILE_S_ISFIFO(mode)	PMEMFILE_S_ISTYPE((mode), PMEMFILE_S_IFIFO)
+#define PMEMFILE_S_ISLNK(mode)	PMEMFILE_S_ISTYPE((mode), PMEMFILE_S_IFLNK)
+#define PMEMFILE_S_ISSOCK(mode)	PMEMFILE_S_ISTYPE((mode), PMEMFILE_S_IFSOCK)
+
+#define PMEMFILE_S_ISUID	04000
+#define PMEMFILE_S_ISGID	02000
+#define PMEMFILE_S_ISVTX	01000
+
+#define PMEMFILE_S_IRUSR	0400
+#define PMEMFILE_S_IWUSR	0200
+#define PMEMFILE_S_IXUSR	0100
+#define PMEMFILE_S_IRWXU	0700
+
+#define PMEMFILE_S_IRGRP	040
+#define PMEMFILE_S_IWGRP	020
+#define PMEMFILE_S_IXGRP	010
+#define PMEMFILE_S_IRWXG	070
+
+#define PMEMFILE_S_IROTH	04
+#define PMEMFILE_S_IWOTH	02
+#define PMEMFILE_S_IXOTH	01
+#define PMEMFILE_S_IRWXO	07
+
+#define PMEMFILE_ACCESSPERMS	0777
+#define PMEMFILE_ALLPERMS		07777
+
+#define PMEMFILE_AT_SYMLINK_NOFOLLOW	0x100
+#define PMEMFILE_AT_REMOVEDIR		0x200
+#define PMEMFILE_AT_SYMLINK_FOLLOW	0x400
+#define PMEMFILE_AT_NO_AUTOMOUNT	0x800
+#define PMEMFILE_AT_EMPTY_PATH		0x1000
+#define PMEMFILE_AT_EACCESS		0x200
+
+#define PMEMFILE_F_DUPFD 0
+#define PMEMFILE_F_GETFD 1
+#define PMEMFILE_F_SETFD 2
+#define PMEMFILE_F_GETFL 3
+#define PMEMFILE_F_SETFL 4
+
+#define PMEMFILE_F_RDLCK  0
+#define PMEMFILE_F_WRLCK  1
+#define PMEMFILE_F_UNLCK  2
+
+#define PMEMFILE_F_GETLK  5
+#define PMEMFILE_F_SETLK  6
+#define PMEMFILE_F_SETLKW 7
+
+#define PMEMFILE_SEEK_SET  0
+#define PMEMFILE_SEEK_CUR  1
+#define PMEMFILE_SEEK_END  2
+#define PMEMFILE_SEEK_DATA 3
+#define PMEMFILE_SEEK_HOLE 4
+
+#define PMEMFILE_DT_UNKNOWN	0
+#define PMEMFILE_DT_FIFO 	1
+#define PMEMFILE_DT_CHR		2
+#define PMEMFILE_DT_DIR		4
+#define PMEMFILE_DT_BLK		6
+#define PMEMFILE_DT_REG		8
+#define PMEMFILE_DT_LNK		10
+#define PMEMFILE_DT_SOCK	12
+#define PMEMFILE_DT_WHT		14
+
+#define PMEMFILE_R_OK 4
+#define PMEMFILE_W_OK 2
+#define PMEMFILE_X_OK 1
+#define PMEMFILE_F_OK 0
+
+#define PMEMFILE_FL_KEEP_SIZE       0x01
+#define PMEMFILE_FL_PUNCH_HOLE      0x02
+#define PMEMFILE_FL_COLLAPSE_RANGE  0x08
+#define PMEMFILE_FL_ZERO_RANGE      0x10
+#define PMEMFILE_FL_INSERT_RANGE    0x20
+
+#define PMEMFILE_MAP_FAILED	((void *) -1)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * opaque type internal to libpmemfile
+ */
+typedef struct pmemfilepool PMEMfilepool;
+typedef struct pmemfile_file PMEMfile;
+
+typedef mode_t pmemfile_mode_t;
+typedef uid_t pmemfile_uid_t;
+typedef gid_t pmemfile_gid_t;
+typedef ssize_t pmemfile_ssize_t;
+#ifndef __off64_t_defined
+#error off64_t is not available, on glibc enable _LARGEFILE64_SOURCE, _XOPEN_SOURCE>=500 or _GNU_SOURCE to define it
+#endif
+typedef off64_t pmemfile_off_t;
+typedef nlink_t pmemfile_nlink_t;
+typedef blksize_t pmemfile_blksize_t;
+typedef blkcnt64_t pmemfile_blkcnt_t;
+typedef dev_t pmemfile_dev_t;
+typedef ino64_t pmemfile_ino_t;
+
+typedef struct timespec pmemfile_timespec_t;
+typedef struct stat pmemfile_stat_t;
+
+#define PMEMFILE_AT_CWD ((PMEMfile *)(((unsigned char *)0) - 1))
+
+PMEMfilepool *pmemfile_mkfs(const char *pathname, size_t poolsize,
```

Comment:
pmemfile_create already exists. I don't see any problem with pmemfile_mkfs, but I don't care - I'll rename pmemfile_mkfs to pmemfile_pool_create.